app/src/pages/loginPage/loginPage.tsx

```diff
@@ -13,20 +12,53 @@ export interface ILoginPageProps {
 export const LoginPage = (props: ILoginPageProps): React.ReactElement => {
   const history = useHistory();
   const [email, setEmail] = React.useState<string | null>(null);
+  const [emailError, setEmailError] = React.useState<string | null>(null);
   const [password, setPassword] = React.useState<string | null>(null);
+  const [passwordError, setPasswordError] = React.useState<string | null>(null);
   const [isLoading, setIsLoading] = React.useState<boolean>(false);
 
-  const onLoginClicked = (): void => {
+  const onLoginClicked = async (): Promise<void> => {
     setIsLoading(true);
-    asyncSleep(1000).then((): void => {
-      // eslint-disable-next-line no-console
-      console.log('email', email, 'password', password);
-    }).catch((error: KibaException): void => {
-      console.error('error', error);
-    });
+    if (!email) {
+      setEmailError('Enter email address');
+    }
+    // Regex copied from https://emailregex.com/
+    const emailRegex = new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
+    if (email && !emailRegex.test(String(email).toLowerCase())) {
+      setEmailError('Please enter a valid email address');
```

Comment:
i don't think this condition is hit. some other thing pops up before any of this code is run in the case of an invalid email address. are you aware of this?